conjure-java/src/main/java/com/palantir/conjure/java/cli/ConjureJavaCli.java

```diff
@@ -127,31 +133,31 @@ public void run() {
         @SuppressWarnings("BanSystemErr")
         public void run() {
             CliConfiguration config = getConfiguration();
-            if (config.generateObjects() && !config.featureFlags().contains(FeatureFlags.UseImmutableBytes)) {
-                System.err.println("[WARNING] Using deprecated ByteBuffer codegen, please enable the "
-                        + "--useImmutableBytes feature flag to opt into the preferred implementation");
-            }
             try {
                 ConjureDefinition conjureDefinition = OBJECT_MAPPER.readValue(config.input(), ConjureDefinition.class);
-                TypeGenerator typeGenerator = new ObjectGenerator(config.featureFlags());
-                ServiceGenerator jerseyGenerator = new JerseyServiceGenerator(config.featureFlags());
-                ServiceGenerator retrofitGenerator = new Retrofit2ServiceGenerator(config.featureFlags());
-                ServiceGenerator undertowGenerator = new UndertowServiceGenerator(config.featureFlags());
-
                 if (config.generateObjects()) {
-                    typeGenerator.emit(conjureDefinition, config.outputDirectory());
+                    if (!config.featureFlags().contains(FeatureFlags.UseImmutableBytes)) {
+                        System.err.println("[WARNING] Using deprecated ByteBuffer codegen, please enable the "
+                                + "--useImmutableBytes feature flag to opt into the preferred implementation");
+                    }
+                    new ObjectGenerator(config.featureFlags()).emit(conjureDefinition, config.outputDirectory());
+                }
+                if (config.generateJaxRsClient()) {
+                    new JaxRsClientGenerator(config.featureFlags()).emit(conjureDefinition, config.outputDirectory());
                 }
                 if (config.generateJersey()) {
-                    jerseyGenerator.emit(conjureDefinition, config.outputDirectory());
+                    new JerseyServiceGenerator(config.featureFlags()).emit(conjureDefinition, config.outputDirectory());
                 }
                 if (config.generateRetrofit()) {
-                    retrofitGenerator.emit(conjureDefinition, config.outputDirectory());
+                    new Retrofit2ServiceGenerator(config.featureFlags())
+                            .emit(conjureDefinition, config.outputDirectory());
                 }
                 if (config.generateUndertow()) {
-                    undertowGenerator.emit(conjureDefinition, config.outputDirectory());
+                    new UndertowServiceGenerator(config.featureFlags())
+                            .emit(conjureDefinition, config.outputDirectory());
                 }
             } catch (IOException e) {
-                throw new RuntimeException("Error parsing definition", e);
+                throw new SafeRuntimeException("Error parsing definition", e);
```

Comment:
What is the purpose of safe exceptions here?

Reply:
We use them elsewhere, and it doesn't produce a different message in this case. I prefer to consistently use safe exception types across our codebases so we can avoid accidentally using the wrong ones where it matters.

In practice this one makes no difference.

Reply:
Mixed feelings about this. Safe imho is a logging concept and should not be used in a non-relevant contexts.

Reply:
Reverting the exception type change because it's not relevant to this PR.